sdk/core/Azure.Core/src/Serialization/IModelJsonSerializable.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Indicates that the implementer can be serialized and deserialized as JSON.
+    /// </summary>
+    /// <typeparam name="T">The type to deserialize the JSON value into.</typeparam>
+    public interface IModelJsonSerializable<out T> : IModelSerializable<T>
+    {
+        /// <summary>
+        /// Serializes the model to the provided <see cref="Utf8JsonWriter"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="Utf8JsonWriter"/> to serialize into.</param>
+        /// <param name="options">The <see cref="ModelSerializerOptions"/> to use.</param>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
```

Comment:
Good question this analyzer was written before my time I don't have the history here.  I believe it was to avoid exposing a public serialization tied to anything since the feature wasn't ready at the time.

Reply:
I believe it is to prevent use of public STJ types outside specific serialization types in Azure.Core.  I think the goal of this is for Azure SDK libraries to remain agnostic about how JSON is created and used so as not to bake dependencies on specific technologies into public interfaces that can't be changed.